assets/js/screens/Collection/CollectionImageModal.jsx

```diff
@@ -0,0 +1,136 @@
+import React, { useState, useEffect } from "react";
+import { useMutation } from "@apollo/react-hooks";
+import { SET_COLLECTION_IMAGE } from "../../components/Collection/collection.query";
+import { toastWrapper } from "../../services/helpers";
+
+const CollectionImageModal = ({ collection, isModalOpen, handleClose }) => {
+  const [selectedWork, setSelectedWork] = useState();
+  const [filteredWorkImages, setFilteredWorkImages] = useState();
+  useEffect(() => {
+    setFilteredWorkImages(collection ? collection.works : []);
+  }, []);
+
+  const handleFilterChange = (e) => {
+    const filterValue = e.target.value.toUpperCase();
+    if (!filterValue) {
+      return setFilteredWorkImages(collection ? collection.works : []);
+    }
+    const filteredList = collection.works.filter((work) => {
+      return work.descriptiveMetadata.title
+        ? work.descriptiveMetadata.title.toUpperCase().indexOf(filterValue) > -1
+        : false;
+    });
+    setFilteredWorkImages(filteredList);
+  };
+
+  let [
+    setCollectionImage,
+    { loading, error: mutationError, data },
+  ] = useMutation(SET_COLLECTION_IMAGE, {
+    onCompleted({ setCollectionImage }) {
```

Comment:
I can't remember, does this fire even if there is a mutation error?   If so, then we should make the Toast message conditional, or put it in some type of on success callback function Apollo Client gives us?

Reply:
That is the callback for success per docs : 
```
onCompleted | (data: TData) => void | A callback executed once your mutation successfully completes
```
I tested this with a random workid and it goes into `onError`. I'll add a toast with failure message for the `onError` condition.